File: sync/src/test/java/org/podval/sync/UriParserTest.java
package org.podval.sync;

import org.podval.photo.ConnectionDescriptor;
import org.kohsuke.args4j.CmdLineException;

import org.junit.Assert;
import org.junit.Test;


public class UriParserTest {

    @Test
    public void schemeHostPath() throws CmdLineException {
        final ConnectionDescriptor result = UriParser.fromUri("scheme://host/path", null);
        Assert.assertEquals("scheme", result.getScheme());
        Assert.assertEquals("host", result.getHost());
        Assert.assertEquals("/path", result.getPath());
    }


    @Test
    public void loginPassword() throws CmdLineException {
        loginPassword("http://a@b.c/d/e", "a", null);
        loginPassword("http://a:b@b.c/d/e", "a", "b");
        loginPassword("http://b.c/d/e", null, null);
        loginPassword("http://a@/d/e", "a", null);
        loginPassword("http://a:p@/d/e", "a", "p");
    }


    private void loginPassword(final String uri, final String login, final String password)
        throws CmdLineException
    {
        final ConnectionDescriptor result = UriParser.fromUri(uri, null);
        Assert.assertEquals(login, result.getLogin());
        Assert.assertEquals(password, result.getPassword());
    }
}

File: sync/src/test/scala/org/podval/photo/cli/UriParserTest.scala
package org.podval.photo.cli

import org.kohsuke.args4j.CmdLineException

import org.junit.{Assert, Test}


class UriParserTest {

    @Test @throws(classOf[CmdLineException])
    def schemeHostPath() {
        val result = UriParser.fromUri("scheme://host/path", null)
        Assert.assertEquals("scheme", result.scheme)
        Assert.assertEquals("host", result.host)
        Assert.assertEquals("/path", result.path)
    }


    @Test @throws(classOf[CmdLineException])
    def loginPassword() {
        loginPassword("http://a@b.c/d/e", Some("a"), None)
        loginPassword("http://a:b@b.c/d/e", Some("a"), Some("b"))
        loginPassword("http://b.c/d/e", None, None)
        loginPassword("http://a@/d/e", Some("a"), None)
        loginPassword("http://a:p@/d/e", Some("a"), Some("p"))
    }


    @throws(classOf[CmdLineException])
    private def loginPassword(uri: String, login: Option[String], password: Option[String]) {
        val result = UriParser.fromUri(uri, null)
        Assert.assertEquals(login, result.login)
        Assert.assertEquals(password, result.password)
    }
}
